### DISCUSSION TITLE
Computing focus setting, and fields of view

### DISCUSSION THREAD

**Discussion by ralatsdc** [AxisCommunications/discussions#198] [VAPIX] [answered]
We are writing a Python application which uses Python bindings to the VAPIX library to configure and control an AXIS M5525-E PTZ Network Camera.

Our application requires:
1) Setting zoom, then disabling autofocus and setting focus knowing distance to an object.
2) Relating the zoom setting to the horizontal and vertical fields of view (FOVs).

We understand how to use the VAPIX library to set the values.

However, we need to understand how to compute
1) The focus setting given the zoom setting, and distance to the object
2) The horizontal and vertical FOVs given the zoom setting

**Comment by whittenator** [comment, marked as answer]
> We are writing a Python application which uses Python bindings to the VAPIX library to configure and control an AXIS M5525-E PTZ Network Camera.
> 
> Our application requires:
> 
> 1. Setting zoom, then disabling autofocus and setting focus knowing distance to an object.
> 2. Relating the zoom setting to the horizontal and vertical fields of view (FOVs).
> 
> We understand how to use the VAPIX library to set the values.
> 
> However, we need to understand how to compute
> 
> 1. The focus setting given the zoom setting, and distance to the object
> 2. The horizontal and vertical FOVs given the zoom setting

This is a very interesting problem set that we are actively trying to solve as well. We are using the Axis Q6215-LE to try and focus on targets are particular distances. We may not have the best solution yet but here our our findings so far:

1. To do a focus-to-distance you will need to create a focus-to-distance interpolation table(This will require you gather zoom, focus, and distance to particular targets at varying distances). Axis doesn't have a direct "focus to distance" command so you will have to come up with an interpolation table that will allow you to give a particular distance to a target with the current zoom level and it will output the focus value you need.(We captured all of this data using the autofocus to get the ideal focus value but we weren't able to come up with a good interpolation table as of yet as the focus/zoom ratio seems to be sporadic).
2. To get horizontal FOV, you can do this in two different ways. The easiest is to ssh into the camera like so:
```bash
ssh root@<your_ip>

# Run the following commands:
statuscacheclient get RAW PTZ_POSITION_IS1.ZOOM_HFOV
statuscacheclient get RAW PTZ_POSITION_IS1.ZOOM_VFOV
```
Now a quick and dirty method for you is to capture the Horizontal/Vertical FOV at the lowest Zoom(1) and the highest Zoom(30 in our case) and then do a ranging method in python like so:
```python
def zoom_to_fov(zoom):
    return 49.138081 - (49.138081 - 1.746173) * ((zoom - 1) / (30 - 1))

for i in range(1, 31):
    print(f"Zoom {i} = {zoom_to_fov(i):.6f} degrees")
```

I know you posted this in March, but have you found anything different than the above?

**Comment by ralatsdc** [comment]
No, we essentially solved this with a simple calibration, much like yours.
